src/main/java/randoop/operation/TypedClassOperation.java

```diff
@@ -174,4 +186,49 @@ public boolean hasWildcardTypes() {
     }
     return new ArrayList<>(paramSet);
   }
+
+  /**
+   * Returns the {@link RawSignature} for this operation if it is a method or constructor call.
+   *
+   * @return the {@link RawSignature} of this method or constructor operation, null if it is another
+   *     kind of operation
+   */
+  public RawSignature getRawSignature() {
+    // XXX Awkward: either refactor operations, or allow RawSignature to represent fields, probably both
+    if (!this.isConstructorCall() && !this.isMethodCall()) {
+      return null;
+    }
+    if (rawSignature == null) {
+      Package classPackage = this.declaringType.getPackage();
+      String packageName = (classPackage != null) ? classPackage.getName() : "";
+      String classname = this.getDeclaringType().getRawtype().getUnqualifiedName();
+      String name =
+          this.getUnqualifiedName().equals("<init>") ? classname : this.getUnqualifiedName();
+
+      Class<?>[] parameterTypes =
+          this.isMethodCall()
+              ? ((MethodCall) getOperation()).getMethod().getParameterTypes()
+              : ((ConstructorCall) getOperation()).getConstructor().getParameterTypes();
+      rawSignature = new RawSignature(packageName, classname, name, parameterTypes);
+    }
+    return rawSignature;
+  }
+
+  /**
+   * Creates an operation with the same name, input types, and output type as this operation, but
+   * having the given type as the owning class.
+   *
+   * <p>Note: this is only a valid object if {@code type} has the method. This is definitely the
+   * case if {@code type} is a subtype of the declaring type of the operation, but this method does
+   * not force that check because we sometimes want to create the operation for superclasses.
+   *
+   * @param type a subtype of the declaring class of this operation to substitute into the
```

Comment:
This requires that the argument is a subtype, but the previous sentence said it wasn't necessarily one.